### ISSUE TITLE
Hiding script attributes

### ISSUE THREAD

**Issue by LeXXik** [playcanvas/editor#278]
Feature request.

Allow hiding / showing script attributes, based on another attribute of the same script. This is a same feature currently implemented in Editor, but allow public use:
```js
this._field('preWarm').parent.hidden = !this._field('loop').value;
```
Perhaps, a script attribute can contain an optional field that would read a value as a boolean from another field and toggle its visibility based on it.

**Comment by Maksims** [comment]
Feels like a case for an Editor Plugin.

**Comment by LeXXik** [comment]
Right, but until we have a channel for distributing the user plugins, feature requests are the only option, I suppose.

**Comment by LeXXik** [comment]
Forum mention: https://forum.playcanvas.com/t/can-we-change-de-script-attributes-by-the-type-of-an-enum/25865/2

**Comment by marklundin** [comment]
For reference, we've got a proposal for this here https://github.com/playcanvas/editor/issues/1196
